CMakeLists.txt

```diff
@@ -166,9 +166,9 @@ if (NJOBS)
   set(CMAKE_JOB_POOL_PRECOMPILE_HEADER custom)
 endif()
 
-# GCC 8.1 is required to support <charconv> header inclusion in base/string_utils.hpp, otherwise 7.0 is sufficient
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.1)
-  message(FATAL_ERROR "Minimum supported g++ version is 8.1 yours is ${CMAKE_CXX_COMPILER_VERSION}")
+# GCC 8.1 is required to support <charconv> header inclusion in base/string_utils.hpp
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10.0)
+  message(FATAL_ERROR "Minimum supported g++ version is 10.0, yours is ${CMAKE_CXX_COMPILER_VERSION}")
```

Comment:
How do we expect to maintain this minimum `10.0`, if we don't even test in the CI with `10.0`?
Next PR going in after this, can add something, which suddenly needs `14.0` and nobody even notices, because that still passes the CI.

Reply:
Suggestion: remove this check altogether.

Reply:
This check will save us and other contributors time debugging build issues when older compilers are used.

We can switch from clang-18 to gcc-10 in the Linux check `Linux no unity build` job.

Reply:
You can express your usage requirements in CMake using [target_compile_features](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html) together with things from [CMAKE_CXX_KNOWN_FEATURES](https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html).
That way you don't need to check anything - CMake will do it for you.

Reply:
What exactly do you propose to add?

Reply:
I don't propose to add anything, I just don't think this check solves any real problem.

Reply:
Waste of time is a _real_ problem, especially at scale.

Reply:
Unfurtunately `CMAKE_CXX_KNOWN_FEATURES`, doesn't have individual feature support for C++20, and even if it had I don't think it could really subsitute actually testing, and currently we only test something that we could call "upper bound" compilers.
I propose to either split or change the [linux-matrix](https://github.com/organicmaps/organicmaps/blob/master/.github/workflows/linux-check.yaml#L96), so it tests lower bound and upper bound compilers, not just the upper bound as it is currently.
In any case line 169  `# GCC 8.1 is required` should be changed in this PR.